packages/wallet-connectors/src/WalletConnect.ts

```diff
@@ -71,77 +73,109 @@ function accountTransactionPayloadToJson(data: AccountTransactionPayload) {
     });
 }
 
+function serializeInitContractParam(
+    initName: string,
+    parameters: SmartContractParameters | undefined,
+    schema: Schema | undefined
+) {
+    if (!parameters) {
+        if (schema) {
+            throw new Error(`schema provided when 'parameters' is undefined`);
+        }
+        // No parameters provided.
+        return toBuffer('');
+    }
+    if (!schema) {
+        throw new Error(`schema not provided when 'parameters' is present`);
+    }
+    switch (schema.kind) {
+        case 'module':
+            return serializeInitContractParameters(initName, parameters, schemaAsBuffer(schema.value), schema.version);
+        case 'parameter':
+            return serializeTypeValue(parameters, schemaAsBuffer(schema.value));
+        default:
+            throw new UnreachableCaseError('schema', schema);
+    }
+}
+
+function serializeUpdateContractMessage(
+    contractName: string,
+    receiveName: string,
+    parameters: SmartContractParameters | undefined,
+    schema: Schema | undefined
+) {
+    if (!parameters) {
+        if (schema) {
+            throw new Error(`schema provided when 'parameters' is undefined`);
+        }
+        // No parameters provided.
+        return toBuffer('');
+    }
+    if (!schema) {
+        throw new Error(`schema not provided when 'parameters' is present`);
+    }
+    switch (schema.kind) {
+        case 'module':
+            return serializeUpdateContractParameters(
+                contractName,
+                receiveName,
+                parameters,
+                schemaAsBuffer(schema.value),
+                schema.version
+            );
+        case 'parameter':
+            return serializeTypeValue(parameters, schemaAsBuffer(schema.value));
+        default:
+            throw new UnreachableCaseError('schema', schema);
+    }
+}
+
+function schemaAsBuffer(schema: string) {
+    return toBuffer(schema, 'base64');
+}
```

Comment:
This function will silently ignore errors in conversion. This is very bad. Especially since neither the name nor the type indicate anything of the sort.

Reply:
How can we check if there are errors?

Reply:
added constructor functions that take the schema as base64 and checks the encoding eagerly, i.e. as early as possible like when the constants are initially created